circle.yml

```diff
@@ -1,28 +1,80 @@
-machine:
-  environment:
-    YARN_VERSION: 0.24.3
-    PATH: "${PATH}:${HOME}/.yarn/bin:${HOME}/${CIRCLE_PROJECT_REPONAME}/node_modules/.bin"
-  node:
-    version: 8
-dependencies:
-  pre:
-    - |
-      if [[ ! -e ~/.yarn/bin/yarn || $(yarn --version) != "${YARN_VERSION}" ]]; then
-        curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version $YARN_VERSION
-      fi
-    - nvm install 4
-    - nvm install 6
+aliases:
+  - &restore-cache
+    keys:
+      - dependencies-{{ .Branch }}-{{ checksum "package.json" }}
+      # Fallback in case checksum fails
+      - dependencies-{{ .Branch }}-
 
-  cache_directories:
-    - ~/.yarn
-    - ~/.cache/yarn
-  override:
-    - yarn install --no-progress
-test:
-  override:
-    - nvm alias default 6
-    - yarn run test-ci
-    - nvm alias default 4
-    - yarn run test-ci
-    - nvm alias default 8
-    - yarn run test-ci
+  - &save-cache
+    paths:
+      - node_modules
+    key: dependencies-{{ .Branch }}-{{ checksum "package.json" }}
+
+  - &deploy
+    command: |
+      npm run prepare-release
+      npm deploy
```

Comment:
shouldn't this be `npm publish`?

Reply:
Also, let's use yarn for running the `prepare-release` script and also run the `cleanup-release` script afterwards:

```sh
yarn run prepare-release
npm publish
yarn run cleanup-release
```

Reply:
> shouldn't this be npm publish?

🤦‍♂️ 

> Also, let's use yarn for running the prepare-release script and also run the cleanup-release script afterwards:

Do we really need a cleanup here?

Reply:
> Do we really need a cleanup here?

In this current scenario is not really needed, but it's still good to leave the repo cleaned up (less confusion for the future engineer that may want to add another step after deploy)